### PR TITLE
Add file open/closed/rotation metrics

### PR DESCRIPTION
**Pull request by mrueg**
This is a non-ruby coder attempt to add these metrics that got requested in 
https://github.com/fluent/fluent-plugin-prometheus/issues/201

@ashie if you have some time to review, that would be appreciated!

### CODE REVIEW

File: lib/fluent/plugin/in_prometheus_tail_monitor.rb [hunk 1]
```diff
@@ -72,9 +81,13 @@ def update_monitor_info
           # Access to internal variable of internal class...
           # Very fragile implementation
           pe = watcher.instance_variable_get(:@pe)
+          monitor_info = watcher.instance_variable_get(:@metrics)
```
Comment (ashie) [inline lib/fluent/plugin/in_prometheus_tail_monitor.rb:84]: Please add a guard for the case `monitor_info == nil`.
Fluentd v1.14.0 or former don't have this feature.
In this case, the plugin will crash.
Users might upgrade this plugins without upgrading Fluentd.

**Review by ashie — CHANGES_REQUESTED** [review]
Sorry for my late response.
Basically it looks good to me but please consider combination with older Fluentd too.

File: lib/fluent/plugin/in_prometheus_tail_monitor.rb (hunk 1 above)
Comment (mrueg) [reply lib/fluent/plugin/in_prometheus_tail_monitor.rb:84]: Thanks for the review, I've added a guard.

File: lib/fluent/plugin/in_prometheus_tail_monitor.rb [hunk 2]
```diff
@@ -51,7 +51,16 @@ def start
...
+        closed_file_metrics: get_gauge(
+          :fluentd_tail_file_closed,
+          'Number of files closed.'),
+        opened_file_metrics: get_gauge(
+          :fluentd_tail_file_opened,
+          'Number of files opened.'),
+        rotated_file_metrics: get_gauge(
+          :fluentd_tail_file_rotated,
+          'Number of files rotated.'),
+    }
```
Comment (ashie) [inline lib/fluent/plugin/in_prometheus_tail_monitor.rb:63]: Please fix indent.

```suggestion
      }
```

File: lib/fluent/plugin/in_prometheus_tail_monitor.rb [hunk 3]
```diff
@@ -72,9 +81,15 @@ def update_monitor_info
           # Access to internal variable of internal class...
           # Very fragile implementation
           pe = watcher.instance_variable_get(:@pe)
+          monitor_info = watcher.instance_variable_get(:@metrics)
           label = labels(info, watcher.path)
           @metrics[:inode].set(pe.read_inode, labels: label)
           @metrics[:position].set(pe.read_pos, labels: label)
+          if monitor_info != nil
```
Comment (ashie) [inline lib/fluent/plugin/in_prometheus_tail_monitor.rb:88]:
```suggestion
          unless monitor_info.nil?
```

**Review by ashie — CHANGES_REQUESTED** [review]
Sorry, there are some more nitpicks.

**Review by ashie — APPROVED** [review]

**Comment by ashie** [comment]
Thanks!

**Pull request merged**